src/main.cpp

```diff
@@ -1468,6 +1501,15 @@ extern "C" {
       }
     }
 
+    if (strcmp(manh, "") == 0)
```

Comment:
This is very confusing for me, after 5 times reading I still don't understand why need 3 strings for this task.

IMO this is the place where:
- make a copy of props.m_strURL
- look if | is in string, if so:
- if inputstream.adaptive.stream_headers is not given, parse it into mahd
- remove the | part from the copy.

Parsing of a string for header parts should be done in a function (parseheader(std::map, const char *)

Doing the parsing stuff should be only done once (here!!), having it in download() means that we parse the same stuff every segment again, this is not very resource friendly.